packages/synthetics-sdk-broken-links/src/link_utils.ts

```diff
@@ -73,13 +75,50 @@ export function setDefaultOptions(
     per_link_options: {},
   };
 
+  const output_options: BrokenLinksResultV1_BrokenLinkCheckerOptions =
+    {} as BrokenLinksResultV1_BrokenLinkCheckerOptions;
   const objKeys = Object.keys(default_options) as Array<
     keyof BrokenLinksResultV1_BrokenLinkCheckerOptions
   >;
   for (const key of objKeys) {
-    if (!(key in options) && key !== 'origin_url') {
-      /* eslint-disable */
-      (options as any)[key] = default_options[key];
-    }
+    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+    (output_options as any)[key] =
+      !(key in options) || key === 'per_link_options'
+        ? default_options[key]
+        : options[key];
   }
+
+  // Convert per_link_options to the appropriate type
+  const perLinkOptions: {
+    [key: string]: BrokenLinksResultV1_BrokenLinkCheckerOptions_PerLinkOption;
+  } = {};
+  for (const [url, perLinkOption] of Object.entries(
+    options.per_link_options || {}
+  )) {
+    perLinkOption.expected_status_code;
+    const expected_status_code = (
+      perLinkOption.expected_status_code !== undefined
+        ? typeof perLinkOption.expected_status_code === 'number'
+          ? { status_value: perLinkOption.expected_status_code }
+          : {
+              status_class:
+                ResponseStatusCode_StatusClass[
+                  perLinkOption.expected_status_code
+                ],
+            }
+        : { status_class: ResponseStatusCode_StatusClass.STATUS_CLASS_2XX }
+    ) as ResponseStatusCode;
```

Comment:
Is it possible to untangle lines 99-110 into multiple if/else blocks or add super detailed comments describing what each portion is doing? nesting multiple ternary operators is pretty hard to read

Reply:
very good point